pkg/api/write.go

```diff
@@ -187,7 +185,7 @@ func decodeSnappy(w http.ResponseWriter, r *http.Request, m *Metrics) bool {
 	return true
 }
 
-func ingest(inserter ingestor.DBInserter, dataParser *parser.DefaultParser) func(http.ResponseWriter, *http.Request, *Metrics) bool {
+func ingest(inserter ingestor.DBInserter, dataParser *parser.DefaultParser, metrics *Metrics) func(http.ResponseWriter, *http.Request, *Metrics) bool {
```

Comment:
do we want to remove the metrics argument from the returned function then? I'm confused why this change was needed but we shouldn't have metrics passed in both ingest and the returned function

```suggestion
func ingest(inserter ingestor.DBInserter, dataParser *parser.DefaultParser, metrics *Metrics) func(http.ResponseWriter, *http.Request) bool {
```

Reply:
The reason metrics is returned to satisfy the return type of `writeStage`. But, since the `var metrics *Metrics` is a singleton in `api` package, we can get rid of these _passing of metrics address_ and use the singleton itself.